odoo/addons/test_lint/tests/_odoo_checker_gettext.py

```diff
@@ -49,10 +50,46 @@ class OdooBaseChecker(BaseChecker):
             'gettext-repr',
             'Don\'t use %r to automatically insert quotes in translation strings. Quotes can be different depending on the language: they must be part of the translated string.',
         ),
+        'E8507': (
+            'Static string passed to %s without gettext call.',
+            'missing-gettext',
+            'Ensure that all static strings passed to certain constructs are wrapped in a gettext call.',
+        ),
     }
 
-    @only_required_for_messages('gettext-variable', 'gettext-placeholders', 'gettext-repr')
+    required_inside_functions = ['UserError', 'ValidationError', 'AccessError', 'AccessDenied', 'MissingError']
+
+    @only_required_for_messages('missing-gettext', 'gettext-variable', 'gettext-placeholders', 'gettext-repr')
     def visit_call(self, node):
+        file_path = self.linter.current_file
+        if (
+            "test" in file_path
+            and (module_path := find_module_root(file_path))
+            and (
+                # Check if the module's base name starts with "test_"
+                # This identifies modules dedicated to testing (e.g., test_mail)
+                os.path.basename(os.path.normpath(module_path)).startswith("test_")
+                # OR check if the file is in a "tests" subdirectory within the module
+                # This identifies test files located within the standard "tests" directory/package
+                or file_path[len(module_path) + 1 :].startswith("tests" + os.sep)
+            )
+        ):
+            return
+
+        if (
+            isinstance(node.func, astroid.Name)
```

Comment:
You may want to write some tests for your check. Consider `exceptions.UserError`: I think `UserError` is an instance of `astroid.Attribute` in that context, and therefore is not caught by your check